ci/docker/serve.build.Dockerfile

```diff
@@ -20,19 +20,13 @@ RUN <<EOF
 
 set -euo pipefail
 
-pip install -U torch==2.0.1 torchvision==0.15.2
-pip install -U tensorflow==2.13.1 tensorflow-probability==0.21.0
 pip install -U --ignore-installed \
   -c python/requirements_compiled.txt \
   -r python/requirements.txt \
   -r python/requirements/test-requirements.txt
-
-# doc requirements
-#
-# TODO (shrekris-anyscale): Remove transformers after core transformer requirement 
-# is upgraded
-pip install transformers==4.30.2
-pip install -c python/requirements_compiled.txt aioboto3
+pip install -U -c python/requirements_compiled.txt \
+  tensorflow tensorflow-probability torch torchvision \
+  transformers aioboto3
```

Comment:
transformers can be removed now?

@shrekris-anyscale ?

Reply:
@can-anyscale The [git blame](https://github.com/ray-project/ray/blame/master/ci/docker/serve.build.Dockerfile#L32) shows that you committed that code. We probably synced about it, but I don't remember why we added that `TODO`. Do you remember?

Reply:
It might be due to #36011. I see that the compiled requirements still use `transformers==4.19.1` ([link](https://github.com/shrekris-anyscale/ray/blob/2d63f638921458f05d0ad9d75bc703e778b178d2/python/requirements_compiled.txt#L2347)). We probably still need the pin; otherwise, this requirement will accept the older version.